mmcls/models/backbones/tnt.py

```diff
@@ -0,0 +1,344 @@
+                 num_heads_in=4,
+                 ffn_ratio=4,
+                 drop_rate=0.,
+                 attn_drop_rate=0.,
+                 drop_path_rate=0.,
+                 num_fcs=2,
+                 qkv_bias=False,
+                 act_cfg=dict(type='GELU'),
+                 norm_cfg=dict(type='LN'),
+                 batch_first=True,
+                 init_cfg=None):
+        super(TnTLayer, self).__init__(init_cfg=init_cfg)
+
+        # Inner transformer
+        self.norm_in = build_norm_layer(norm_cfg, inner_dims)[1]
+        self.attn_in = MultiheadAttention(
+            embed_dims=inner_dims,
+            num_heads=num_heads_in,
+            attn_drop=attn_drop_rate,
+            proj_drop=drop_rate,
+            dropout_layer=dict(type='DropPath', drop_prob=drop_path_rate),
+            batch_first=batch_first)
+
+        self.norm_ffn_in = build_norm_layer(norm_cfg, inner_dims)[1]
+        self.ffn_in = FFN(
+            embed_dims=inner_dims,
+            feedforward_channels=inner_dims * ffn_ratio,
+            num_fcs=num_fcs,
+            ffn_drop=drop_rate,
+            dropout_layer=dict(type='DropPath', drop_prob=drop_path_rate),
+            act_cfg=act_cfg)
+
+        self.norm_proj = build_norm_layer(norm_cfg, inner_dims)[1]
+        self.projection = nn.Linear(
+            inner_dims * num_pixel, embed_dims, bias=True)
+
+        # Outer transformer
+        self.norm_out = build_norm_layer(norm_cfg, embed_dims)[1]
+        self.attn_out = MultiheadAttention(
+            embed_dims=embed_dims,
+            num_heads=num_heads_out,
+            attn_drop=attn_drop_rate,
+            proj_drop=drop_rate,
+            dropout_layer=dict(type='DropPath', drop_prob=drop_path_rate),
+            batch_first=batch_first)
+
+        self.norm_ffn_out = build_norm_layer(norm_cfg, embed_dims)[1]
+        self.ffn_out = FFN(
+            embed_dims=embed_dims,
+            feedforward_channels=embed_dims * ffn_ratio,
+            num_fcs=num_fcs,
+            ffn_drop=drop_rate,
+            dropout_layer=dict(type='DropPath', drop_prob=drop_path_rate),
+            act_cfg=act_cfg)
+
+        if not qkv_bias:
+            self.attn_in.attn.in_proj_bias = None
+            self.attn_out.attn.in_proj_bias = None
+
+    def forward(self, pixel_embed, patch_embed):
+        # inner
+        pixel_embed = self.attn_in(
+            self.norm_in(pixel_embed), identity=pixel_embed)
+        pixel_embed = self.ffn_in(
+            self.norm_ffn_in(pixel_embed), identity=pixel_embed)
+
+        # outer
+        B, N, C = patch_embed.size()
+        patch_embed[:, 1:] = patch_embed[:, 1:] + self.projection(
+            self.norm_proj(pixel_embed).reshape(B, N - 1, -1))
+        patch_embed = self.attn_out(
+            self.norm_out(patch_embed), identity=patch_embed)
+        patch_embed = self.ffn_out(
+            self.norm_ffn_out(patch_embed), identity=patch_embed)
+
+        return pixel_embed, patch_embed
```

Comment:
If the "inner transformer block" and "outer transformer block" structures are almost the same, can we encapsulate them as one module?
```suggestion
    def forward(self, pixel_embed, patch_embed):
        pixel_embed = self.inner_block(pixel_embed)

        B, N, C = patch_embed.size()
        patch_embed[:, 1:] = patch_embed[:, 1:] + self.projection(
            self.norm_proj(pixel_embed).reshape(B, N - 1, -1))
            
        patch_embed = self.outer_block(patch_embed)

        return pixel_embed, patch_embed
```